src/main/java/net/snowflake/ingest/streaming/internal/BlobBuilder.java

```diff
@@ -83,25 +87,29 @@ static <T> Blob constructBlobAndMetadata(
           flusher.serialize(channelsDataPerTable, filePath);
 
       if (!serializedChunk.channelsMetadataList.isEmpty()) {
-        ByteArrayOutputStream chunkData = serializedChunk.chunkData;
-        Pair<byte[], Integer> paddedChunk =
-            padChunk(chunkData, Constants.ENCRYPTION_ALGORITHM_BLOCK_SIZE_BYTES);
-        byte[] paddedChunkData = paddedChunk.getFirst();
-        int paddedChunkLength = paddedChunk.getSecond();
+        byte[] chunkData = serializedChunk.chunkData.toByteArray();
```

Comment:
this allocation introduces unnecessary GC pressure  when disableChunkEncryption is false. please do this in an else block only when encryption is disabled.